--- a/Sources/Client/Client_Input.cpp
+++ b/Sources/Client/Client_Input.cpp
@@ -38,6 +38,7 @@
 
 #include "NetClient.h"
 
+using namespace std;
 
 SPADES_SETTING(cg_mouseSensitivity, "1");
 SPADES_SETTING(cg_zoomedMouseSensScale, "0.6");
--- a/Sources/Client/Corpse.cpp
+++ b/Sources/Client/Corpse.cpp
@@ -27,6 +27,8 @@
 #include "../Core/Debug.h"
 #include "../Core/Settings.h"
 
+using namespace std;
+
 SPADES_SETTING(r_corpseLineCollision, "1");
 
 namespace spades {
--- a/Sources/Draw/SWMapRenderer.cpp
+++ b/Sources/Draw/SWMapRenderer.cpp
@@ -31,6 +31,8 @@
 #include "SWUtils.h"
 #include <cstdint>
 
+using namespace std;
+
 SPADES_SETTING(r_swUndersampling, "0");
 
 namespace spades {
