--- a/gtk-2.0/gtkrc
+++ b/gtk-2.0/gtkrc
@@ -3,6 +3,7 @@
 # License: GPL-3.0+
 
 gtk-color-scheme = "base_color:#fff\nfg_color:#555\ntooltip_fg_color:#fff\nselected_bg_color:#569be4\nselected_fg_color:#fff\ntext_color:#333\nbg_color:#eee\ntooltip_bg_color:#000\nlink_color:#569be4"
+gtk-icon-sizes = "gtk-large-toolbar=24,24:gtk-small-toolbar=16,16:gtk-button=16,16:gtk-dialog=48,48:panel-menu=16,16:panel=16,16:gtk-menu=16,16:gtk-dnd=48,48"
 
 style "default" {
 	xthickness = 1
