Index: gnucash/trunk/src/gnome-utils/gnc-date-edit.c
===================================================================
--- a/gnucash/trunk/src/gnome-utils/gnc-date-edit.c
+++ b/gnucash/trunk/src/gnome-utils/gnc-date-edit.c
@@ -748,4 +748,4 @@
 {
     GNCDateEdit *gde = data;
-    G_CONST_RETURN char *string;
+    const char *string;
     struct tm tm;

Index: gnucash/trunk/src/import-export/aqbanking/gnc-ab-kvp.c
===================================================================
--- a/gnucash/trunk/src/import-export/aqbanking/gnc-ab-kvp.c
+++ b/gnucash/trunk/src/import-export/aqbanking/gnc-ab-kvp.c
@@ -46,5 +46,5 @@
 static kvp_frame *gnc_ab_get_book_kvp(QofBook *b, gboolean create);
 
-G_CONST_RETURN gchar *
+const gchar *
 gnc_ab_get_account_accountid(const Account *a)
 {
@@ -65,5 +65,5 @@
 }
 
-G_CONST_RETURN gchar *
+const gchar *
 gnc_ab_get_account_bankcode(const Account *a)
 {

Index: gnucash/trunk/src/libqof/qof/guid.c
===================================================================
--- a/gnucash/trunk/src/libqof/qof/guid.c
+++ b/gnucash/trunk/src/libqof/qof/guid.c
@@ -72,5 +72,5 @@
  * @return the value stored in @a value
  */
-G_CONST_RETURN GncGUID*
+const GncGUID*
 gnc_value_get_guid (const GValue *value)
 {

Index: gnucash/trunk/src/libqof/qof/guid.h
===================================================================
--- a/gnucash/trunk/src/libqof/qof/guid.h
+++ b/gnucash/trunk/src/libqof/qof/guid.h
@@ -65,5 +65,5 @@
 
 GType gnc_guid_get_type (void);
-G_CONST_RETURN GncGUID* gnc_value_get_guid (const GValue *value);
+const GncGUID* gnc_value_get_guid (const GValue *value);
 
 /** number of characters needed to encode a guid as a string
