--- GNUmakefile.slitaz	2017-02-22 10:45:30.000000000 +0100
+++ GNUmakefile	2019-03-13 13:44:54.492590684 +0100
@@ -7,7 +7,7 @@ CFLAGS	+= -DVERSION='"$(VERSION)"'
 
 # build
 TARGETS	:= input-events input-kbd input-send input-recv
-TARGETS	+= lsinput emulate-key lircd.conf
+TARGETS	+= lsinput input-recv
 HEADERS	:= EV.h REL.h ABS.h MSC.h LED.h SND.h REP.h KEY.h BTN.h BUS.h SW.h
 
 # default target
