--- src/glade_support.c	Fri Mar 16 15:25:49 2007
+++ src/glade_support.c	Mon Apr  6 17:08:47 2009
@@ -37,10 +37,10 @@
  *                                                                       *
  *                                                                       *
  *************************************************************************/
-typedef struct signal {
+typedef struct _signal {
 	gchar     *name;
 	GCallback callback;
-} signal;
+} _signal;
 
 /*
 ** Signal handler callbascks.
@@ -206,7 +206,7 @@
 static gboolean
 find_and_connect_handler(
 		GtkWidget *widget,
-		signal *signals,
+		_signal *signals,
 		const gchar *signal_name,
 		const gchar *handler_name)
 {
@@ -288,7 +288,7 @@
 		gpointer user_data)
 {
 	gint n;
-	signal entry_signals[] = {
+	_signal entry_signals[] = {
 		{ "activate",           (GCallback)on_any_entry_almost_any },
 		{ "backspace",          (GCallback)on_any_entry_almost_any },
 		{ "copy-clipboard",     (GCallback)on_any_entry_almost_any },
@@ -319,7 +319,7 @@
 		gpointer user_data)
 {
 	variable *var;
-	signal widget_signals[] = {
+	_signal widget_signals[] = {
 		{ "accel-closures-changed",    (GCallback)on_any_widget_almost_any },
 		{ "composited-changed",        (GCallback)on_any_widget_almost_any },
 		{ "grab-focus",                (GCallback)on_any_widget_almost_any },
