--- mtools-3.9.11.ori/Makefile	Tue Sep 16 22:01:39 2008
+++ mtools-3.9.11/Makefile	Tue Sep 16 22:05:15 2008
@@ -216,8 +216,8 @@
 uninstall-info:
 	cd $(DESTDIR)$(infodir) && rm -f mtools.info*
 
-install:	$(DESTDIR)$(bindir)/mtools $(DESTDIR)$(bindir)/floppyd $(DESTDIR)$(bindir)/floppyd_installtest install-man install-links \
-		$(DESTDIR)$(bindir)/mkmanifest install-scripts install-info
+install:	$(DESTDIR)$(bindir)/mtools $(DESTDIR)$(bindir)/floppyd $(DESTDIR)$(bindir)/floppyd_installtest install-links \
+		$(DESTDIR)$(bindir)/mkmanifest install-scripts
 
 uninstall:	uninstall-bin uninstall-man uninstall-links \
 		uninstall-scripts
