--- pcmanfm-0.5.ori/src/ptk/ptk-file-archiver.c	Sun May 25 14:59:12 2008
+++ pcmanfm-0.5/src/ptk/ptk-file-archiver.c	Thu Jun 25 23:28:10 2009
@@ -5,11 +5,13 @@
 *
 *
 * Author: Hong Jen Yee (PCMan) <pcman.tw (AT) gmail.com>, (C) 2006
-*
+*			
 * Copyright: See COPYING file that comes with this distribution
 *
 */
 
+/* 06-25-2009 Erjo: Add support for tazpkg */
+
 #include <glib/gi18n.h>
 #include <string.h>
 
@@ -55,6 +57,12 @@
             "tar -xvf",
             ".tar", TRUE
         },
+		{
+            "application/x-tazpkg",
+            "tazpkg pack",
+            "tazpkg extract",
+            ".tazpkg", FALSE
+        }
         /*
         {
             "application/x-rar",
