--- a/src/defines.h
+++ b/src/defines.h
@@ -4,7 +4,7 @@
 #define UI_SPACING	5
 #define UI_WINDOW_BORDER	10
 
-#define DEFAULT_FILEMANAGER	"nautilus"
+#define DEFAULT_FILEMANAGER	"pcmanfm"
 
 #include "../config.h"
 #include "../lib/gettext.h"

--- a/po/LINGUAS
+++ b.po/LINGUAS
@@ -1 +1 @@
-en@quot en@boldquot fr en_GB de_DE pt_BR ru_RU tr pl es_ES it
+en@quot en@boldquot fr en_GB de_DE pt_BR ru tr pl es_ES it

--- a/cryptkeeper.desktop
+++ b/cryptkeeper.desktop
@@ -3,7 +3,7 @@
 Name=Cryptkeeper
 Exec=cryptkeeper
 Icon=cryptkeeper
-Categories=Application;System;
+Categories=Application;System;Security;
 Terminal=false
 Type=Application
 StartupNotify=false
