--- a/data/leafpad.desktop.in
+++ b/data/leafpad.desktop.in
@@ -1,10 +1,9 @@
 [Desktop Entry]
-Encoding=UTF-8
+Type=Application
 _Name=Leafpad
 _Comment=Simple text editor
+Terminal=false
 Exec=leafpad %f
-Icon=leafpad
+Icon=accessories-text-editor
-Terminal=false
-Type=Application
-MimeType=text/plain
 Categories=GTK;Utility;TextEditor;
+MimeType=text/plain;
--- a/src/callback.c
+++ b/src/callback.c
@@ -336,7 +336,7 @@
 		"authors", authors,
 		"artists", artists,
 		"translator-credits", translator_credits,
-		"logo-icon-name", PACKAGE,
+		"logo-icon-name", "accessories-text-editor",
 		NULL);
 #else
 	static GtkWidget *about = NULL;
--- a/src/window.c
+++ b/src/window.c
@@ -52,7 +52,7 @@
 	gtk_window_set_default_icon(icon);
 */
 #if GTK_CHECK_VERSION(2, 6, 0)
-	gtk_window_set_default_icon_name(PACKAGE);
+	gtk_window_set_default_icon_name("accessories-text-editor");
 #else
  	GdkPixbuf *icon = gdk_pixbuf_new_from_file(
 		ICONDIR G_DIR_SEPARATOR_S PACKAGE ".png", NULL);
