--- src/v4l2_formats.c.old
+++ src/v4l2_formats.c
@@ -20,10 +20,7 @@
 #                                                                               #
 ********************************************************************************/
 
-#include <glib/gprintf.h>
-#include <glib/gstdio.h>
-#include <glib/gutils.h>
-#include <glib/gtestutils.h>
+#include <glib.h>
 #include <fcntl.h>
 #include <string.h>
 #include <sys/types.h>
