--- gpio.c
+++ gpio.slitaz.c
@@ -74,26 +74,6 @@
 	      "       gpio gbr <channel>\n"
 	      "       gpio gbw <channel> <value>" ;	// No trailing newline needed here.
 
-
-/*
- * decodePin:
- *	Decode a pin "number" which can actually be a pin name to represent
- *	one of the Pi's on-board pins.
- *********************************************************************************
- */
-
-static int decodePin (const char *str)
-{
-
-// The first case - see if it's a number:
-
-  if (isdigit (str [0]))
-    return atoi (str) ;
-
-  return 0 ;
-}
-
-
 /*
  * changeOwner:
  *	Change the ownership of the file to the real userId of the calling
