--- childsplay_plugins_lfc-0.90/install.sh.orig	Sat Aug 25 07:53:57 2007
+++ childsplay_plugins_lfc-0.90/install.sh	Thu Jun 11 21:07:43 2009
@@ -8,8 +8,15 @@
 # However, you can change the "prefix" to "/usr" and childsplay will still
 # work. Set it to anything else and your on your own.
   
-PREFIX=/usr/local
+PREFIX=/usr
+PKG_PREFIX=$PWD/_pkg/usr
 
+# Create directories to install the package
+mkdir -p $PKG_PREFIX/lib/games/childsplay/lib
+mkdir -p $PKG_PREFIX/share/assetml
+mkdir -p $PKG_PREFIX/share/childsplay/Data/icons
+mkdir -p $PKG_PREFIX/share/childsplay/lib
+
 #################################################################
 # DON'T EDIT BEHIND THIS POINT 
 #################################################################
@@ -19,7 +26,7 @@
 fi
 
 LOCALEDIR=$PREFIX/share/locale
-ASSETMLDIR=/usr/share/assetml
+ASSETMLDIR=$PREFIX/share/assetml
 PYTHON=`which python`
 SCOREDIR=/var/games
 DOCDIR=$PREFIX/share/doc/childsplay
@@ -30,10 +37,17 @@
 SHARELIBDATADIR=$SHAREDIR/lib
 SHAREDATADIR=$SHAREDIR/Data
 
+# package directories
+PKG_ASSETMLDIR=$PKG_PREFIX/share/assetml
+PKG_CPDIR=$PKG_PREFIX/lib/games/childsplay
+PKG_SHAREDIR=$PKG_PREFIX/share/childsplay
+PKG_SHARELIBDATADIR=$PKG_SHAREDIR/lib
+PKG_SHAREDATADIR=$PKG_SHAREDIR/Data
+
 ##################################################
 #  Package text to be displayed as last step
 # 0 = no readmes to display, 1 = there are readmes
-README=1
+README=0
 # if README=1 then READMES=names of readmes in CWD
 # like this READMES="README README2 README3"
 READMES="README"
@@ -52,8 +66,8 @@
 echo -e "\n>>>>>>>>>>> Install childsplay plugins >>>>>>>>>>>>>>>>>>>>"
 echo -e "\n This release depends on childsplay version >= $DEPEN"
 echo -n " Checking, version = "
-#CP=`which childsplay`
-CP=$BINDIR/childsplay
+CP=`which childsplay`
+#CP=$BINDIR/childsplay
 VERSION=$($CP --version)
 echo $VERSION
 
@@ -73,21 +87,21 @@
 
 set -e
 
-echo -e "\n The path to install the plugins in is $CPDIR."
-echo " If you have installed childsplay in a other place, you can"
-echo " give the full path to childsplay, otherwise just hit enter"
-echo " Hit enter to install in $CPDIR, or give the full path"
-read path
-if [ $path ];then
-    if [ -e $path/install.py ];then
-        CPDIR=path
-    else
-        echo "*********** WARNING *************"
-        echo " $path does not exists or is not the childsplay directory"
-        echo " exit"
-        exit 1
-    fi
-fi
+# echo -e "\n The path to install the plugins in is $CPDIR."
+# echo " If you have installed childsplay in a other place, you can"
+# echo " give the full path to childsplay, otherwise just hit enter"
+# echo " Hit enter to install in $CPDIR, or give the full path"
+# read path
+# if [ $path ];then
+#     if [ -e $path/install.py ];then
+#         CPDIR=path
+#     else
+#         echo "*********** WARNING *************"
+#         echo " $path does not exists or is not the childsplay directory"
+#         echo " exit"
+#         exit 1
+#     fi
+# fi
 echo " Installing in $CPDIR"
 
 echo "Compiling modules to bytecode..."
@@ -97,16 +111,16 @@
 for DIR in `ls $CWD/lib`
 do 
     if [ -d $CWD/lib/$DIR ]
-	 then cp -rf $CWD/lib/$DIR $SHARELIBDATADIR
-	 else cp -f $CWD/lib/$DIR $CPDIR/lib
+	 then cp -rf $CWD/lib/$DIR $PKG_SHARELIBDATADIR
+	 else cp -f $CWD/lib/$DIR $PKG_CPDIR/lib
     fi
 done
 
 echo " Copy misc. data files"
-cp -rf $CWD/Data/*.icon.png $SHAREDATADIR/icons/
+cp -rf $CWD/Data/*.icon.png $PKG_SHAREDATADIR/icons/
 
 echo " Copy assetml files"
-cp -rf $CWD/assetml/* $ASSETMLDIR
+cp -rf $CWD/assetml/* $PKG_ASSETMLDIR
 
 echo -e "\n Everything installed, enjoy\n"
 echo -e "\n***************** IMPORTANT *************************"
