--- lxqtpower/lxqtpowerproviders.cpp
+++ lxqtpower/lxqtpowerproviders.cpp
@@ -497,2 +497,10 @@
                             PowerProvider::DontCheckDBUS);
+        case Power::PowerReboot:
+            return dbusCall(LXQT_SERVICE, LXQT_PATH, LXQT_SERVICE,
+                            QDBusConnection::sessionBus(), "canReboot",
+                            PowerProvider::DontCheckDBUS);
+        case Power::PowerShutdown:
+            return dbusCall(LXQT_SERVICE, LXQT_PATH, LXQT_SERVICE,
+                            QDBusConnection::sessionBus(), "canPoweroff",
+                            PowerProvider::DontCheckDBUS);
         default:
@@ -511,8 +519,2 @@
         command = "logout";
-        break;
-
-    default:
-        return false;
-    }
-
     return dbusCall(LXQT_SERVICE,
@@ -523,2 +525,14 @@
             );
+        break;
+    case Power::PowerReboot:
+        return QProcess::startDetached("reboot || reboot -f");
+        break;
+    case Power::PowerShutdown:
+        return QProcess::startDetached("poweroff");
+        break;
+
+    default:
+        return false;
+    }
+
 }
