--- yasr.h.old	Thu Jan 31 12:56:45 2008
+++ yasr.h	Sat Jan 30 11:37:31 2010
@@ -309,7 +309,9 @@
 extern void opt_write(FILE * fp);
 
 /* openpty.c prototypes */
+#ifndef HAVE_OPENPTY
 extern int openpty(int *, int *, char *, struct termios *, struct winsize *);
+#endif
 
 /* cfmakeraw.c prototypes */
 extern void cfmakeraw(struct termios *);
@@ -318,7 +320,9 @@
 extern int login_tty(int);
 
 /* forkpty.c prototypes */
+#ifndef HAVE_FORKPTY
 extern int forkpty(int *, char *, struct termios *, struct winsize *);
+#endif
 
 /* tbc - Would it be more efficient to ensure that "blank" grids always held
    ascii 0x20 rather than ascii 0x00? */
