--- kazehakase-0.5.8-svn3873_trunk/src/kz-app.c.gnutls3	2009-09-14 01:51:10.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/src/kz-app.c	2013-02-09 21:18:04.000000000 +0900
@@ -494,7 +494,9 @@
 
 #ifdef USE_SSL
         /* initialize gnutls. this function should be called once. */
+#if GNUTLS_VERSION_NUMBER <= 0x020b00
 	gcry_control(GCRYCTL_SET_THREAD_CBS, &gcry_threads_pthread);
+#endif
 	gnutls_global_init();
 #endif
 
