--- kazehakase-0.5.8-svn3873_trunk/src/kz-window.c.default	2010-05-04 00:38:44.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/src/kz-window.c	2012-04-04 12:45:05.443193500 +0900
@@ -830,15 +830,15 @@
 	if (engine)
 	{
 		if (engine[0])
-			web = KZ_WEB(kz_web_new(engine, "gecko"));
+			web = KZ_WEB(kz_web_new(engine, "webkit_gtk"));
 		g_free(engine);
 	}
 
 	if (!web)
-		web = KZ_WEB(kz_web_new("gecko", "gecko"));
-	if (!web)
 		web = KZ_WEB(kz_web_new("webkit_gtk", "webkit_gtk"));
 	if (!web)
+		web = KZ_WEB(kz_web_new("gecko", "gecko"));
+	if (!web)
 		web = KZ_WEB(kz_web_new("ie", "ie"));
 
 	return web;
