--- kazehakase-0.5.8-svn3873_trunk/module/search/kz-hyper-estraier-search.c.egg	2009-02-27 13:28:44.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/module/search/kz-hyper-estraier-search.c	2012-04-04 14:12:59.588721573 +0900
@@ -401,7 +401,7 @@
 					       strlen(G_DIR_SEPARATOR_S));
 
 		thumb_filename =
-			egg_pixbuf_get_thumb_filename(uri,
+			egg_pixbuf_get_thumbnail_filename(uri,
 						      EGG_PIXBUF_THUMBNAIL_LARGE);
 		thumb_uri = g_strdup_printf("history-search:?image=%s",
 					    thumb_filename);
--- kazehakase-0.5.8-svn3873_trunk/module/search/rast-search.c.egg	2009-01-28 04:25:06.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/module/search/rast-search.c	2012-04-04 14:12:53.801803391 +0900
@@ -186,7 +186,7 @@
 			gchar *summary = get_value(line);
 
 			desc = remove_tag(summary, g_strlen(summary));
-			thumb_filename = egg_pixbuf_get_thumb_filename(uri,
+			thumb_filename = egg_pixbuf_get_thumbnail_filename(uri,
 								       EGG_PIXBUF_THUMB_LARGE);
 			thumb_uri = g_strdup_printf("history-search:?image=%s",
 						    thumb_filename);
