--- commoncpp2-1.7.3.org/src/cidr.cpp	2009-01-16 07:27:00.000000000 -0500
+++ commoncpp2-1.7.3/src/cidr.cpp	2009-07-28 15:29:02.881384386 -0400
@@ -199,11 +199,7 @@ void IPV4Cidr::set(const char *cp)
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
 	
-#if defined(_MSC_VER) && _MSC_VER >= 1500
-	ep = (char *)strchr(cp, '/');
-#else
- 	ep = strchr(cp, '/');
-#endif
+ 	ep = (char *)strchr(cp, '/'); // see bug #279505 and #275750
 
 	if(ep)
 		*ep = 0;
@@ -332,7 +328,7 @@ void IPV6Cidr::set(const char *cp)
 	memset(&netmask, 0, sizeof(netmask));
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
-	ep = strchr(cp, '/');
+ 	ep = (char *)strchr(cp, '/'); // see bug #279505 and #275750
 	if(ep)
 		*ep = 0;
 
