--- slim-1.3.1/app.cpp	Fri Sep 26 00:54:15 2008
+++ slim-1.3.1nopwd/app.cpp	Mon May  3 19:44:41 2010
@@ -413,6 +413,7 @@
             case Panel::Exit:
             case Panel::Console:
                 return true; // <--- This is simply fake!
+         
             default:
                 break;
         };
@@ -434,6 +435,11 @@
             case Panel::Console:
                 cerr << APPNAME << ": Got a special command (" << LoginPanel->GetName() << ")" << endl;
                 return true; // <--- This is simply fake!
+	    case Panel::Suspend:
+            case Panel::Halt:
+            case Panel::Reboot:
+                if (cfg->getOption("root_password") == "false")
+                  return true;
             default:
                 break;
         }
--- slim-1.3.1/cfg.cpp	Fri Sep 26 00:54:15 2008
+++ slim-1.3.1nopwd/cfg.cpp	Sun May  2 23:36:46 2010
@@ -37,6 +37,7 @@
     options.insert(option("login_cmd","exec /bin/bash -login ~/.xinitrc %session"));
     options.insert(option("halt_cmd","/sbin/shutdown -h now"));
     options.insert(option("reboot_cmd","/sbin/shutdown -r now"));
+    options.insert(option("root_password","true"));
     options.insert(option("suspend_cmd",""));
     options.insert(option("sessionstart_cmd",""));
     options.insert(option("sessionstop_cmd",""));
Common subdirectories: slim-1.3.1/themes and slim-1.3.1nopwd/themes
