--- resources.qrc
+++ resources.qrc.n
@@ -64,5 +64,25 @@
         <file>images/badge3.png</file>
         <file>images/badge4.png</file>
         <file>sounds/snapshot.wav</file>
+        <file>images/audio-volume-high.png</file>
+        <file>images/audio-volume-muted.png</file>
+        <file>images/bookmark-new.png</file>
+        <file>images/bookmark-remove.png</file>
+        <file>images/document-save.png</file>
+        <file>images/go-next.png</file>
+        <file>images/go-previous.png</file>
+        <file>images/go-top.png</file>
+        <file>images/media-playback-pause.png</file>
+        <file>images/media-playback-start.png</file>
+        <file>images/media-playback-stop.png</file>
+        <file>images/media-skip-backward.png</file>
+        <file>images/media-skip-forward.png</file>
+        <file>images/system-search.png</file>
+        <file>images/video-display.png</file>
+        <file>images/view-fullscreen.png</file>
+        <file>images/view-list-symbolic.png</file>
+        <file>images/view-refresh.png</file>
+        <file>images/view-restore.png</file>
+        <file>images/window-close.png</file>
     </qresource>
 </RCC>
