--- midori/midori-websettings.c	Tue Dec 15 22:34:29 2009
+++ midori/midori-websettings.c	Sun Jan  3 17:20:25 2010
@@ -596,13 +596,12 @@
                                      MIDORI_STARTUP_LAST_OPEN_PAGES,
                                      flags));
 
-    g_object_class_install_property (gobject_class,
                                      PROP_HOMEPAGE,
                                      g_param_spec_string (
                                      "homepage",
                                      _("Homepage:"),
                                      _("The homepage"),
-                                     "http://www.google.com",
+                                     "file:///usr/share/webhome/index.html",
                                      flags));
 
     /**
