--- fs/etc/xdg/midori/search
+++ fs/etc/xdg/midori/search
@@ -1,7 +1,7 @@
 [Duck Duck Go]
 name=Duck Duck Go
 text=Privacy-aware Web Search
-uri=https://duckduckgo.com/?q=%s
+uri=https://duckduckgo.com/?q=%s&t=slitaz
 token=dd
 
 [Yahoo]
@@ -34,3 +34,14 @@
 uri=http://translate.google.com/?q=
 token=gt
 
+[Twitter]
+name=Twitter
+text=Twitter Search
+uri=https://twitter.com/#!/search/realtime/%s
+token=tt
+
+[SliTaz]
+name=SliTaz
+text=SliTaz Search
+uri=http://www.google.com/cse?cx=000868395082919927601:nddq7yjdcxg&cof=FORID:0&q=%s
+token=s
