--- tuxmath_w_fonts-1.7.2/src/SDL_extras.c.old	Mon Mar  9 10:44:42 2009
+++ tuxmath_w_fonts-1.7.2/src/SDL_extras.c	Sat Jun  6 18:55:22 2009
@@ -987,6 +987,7 @@
 /* font in memory once loaded until cleanup.                  */
 static TTF_Font* get_font(int size)
 {
+  static char prev_font_name[FONT_NAME_LENGTH];
   if (size < 0)
   {
     fprintf(stderr, "Error - requested font size %d is negative\n", size);
@@ -1001,7 +1002,7 @@
   }
 
   /* If the font has changed, we need to wipe out the old ones: */
-  if (0 != strncmp(prev_font_name, Opts_FontName(),sizeof(prev_font_name)))
+  if (0 != strncmp(prev_font_name, Opts_FontName(), FONT_NAME_LENGTH))
   {
     free_font_list();
     strncpy(prev_font_name, Opts_FontName(), sizeof(prev_font_name));
