--- a/gtk-2.0/gtkrc
+++ b/gtk-2.0/gtkrc
@@ -17,7 +17,7 @@
 include "bars.rc"        # Styles the menubar.
 
 ### ICONS ###
-gtk-icon-sizes = "gtk-button=16,16" # This makes button icons smaller.
+gtk-icon-sizes = "gtk-large-toolbar=24,24:gtk-small-toolbar=16,16:gtk-button=16,16:gtk-dialog=48,48:panel-menu=16,16:panel=16,16:gtk-menu=16,16:gtk-dnd=48,48"
 
 ### MISC ###
 gtk-button-images	= 1 # Enables icons in buttons
