--- 915resolution.c	
+++ 915resolution.c	
@@ -56,12 +56,12 @@
 
 typedef enum {
     CT_UNKWN, CT_500GMA, CT_830, CT_845G, CT_855GM, CT_865G, CT_915G, CT_915GM, CT_945G, CT_945GM,
-    CT_945GME, CT_946GZ, CT_G965, CT_Q965, CT_965GM
+    CT_945GME, CT_946GZ, CT_G965, CT_Q965, CT_965GM, CT_Cedarview
 } chipset_type;
 
 char * chipset_type_names[] = {
     "UNKNOWN", "500GMA", "830", "845G", "855GM", "865G", "915G", "915GM", "945G", "945GM",
-    "945GME", "946GZ", "G965", "Q965", "965GM"
+    "945GME", "946GZ", "G965", "Q965", "965GM", "Cedarview"
 };
 
 typedef enum {
@@ -237,6 +237,10 @@
         type = CT_965GM;
         break;    
 
+    case 0xbf38086:
+	type = CT_Cedarview;
+        break;    
+
     default:
         type = CT_UNKWN;
         break;
@@ -526,6 +530,7 @@
     case CT_G965:
     case CT_Q965:
     case CT_965GM:
+    case CT_Cedarview:
         outl(0x80000090, 0xcf8);
         map->b1 = inb(0xcfd);
         map->b2 = inb(0xcfe);
@@ -569,6 +574,7 @@
     case CT_G965:
     case CT_Q965:
     case CT_965GM:
+    case CT_Cedarview:
         outl(0x80000090, 0xcf8);
         outb(map->b1, 0xcfd);
         outb(map->b2, 0xcfe);
@@ -837,6 +843,9 @@
         }
         else if (!strcmp(argv[index], "965GM")) {
             *forced_chipset = CT_965GM;
+        }
+        else if (!strcmp(argv[index], "Cedarview")) {
+            *forced_chipset = CT_Cedarview;
         }
         else {
             *forced_chipset = CT_UNKWN;
