--- liblognorm-0.3.2-ori/src/Makefile.am
+++ liblognorm-0.3.2/src/Makefile.am
@@ -9,7 +9,7 @@
 bin_PROGRAMS = normalizer
 normalizer_SOURCES = normalizer.c
 normalizer_CPPFLAGS =  -I$(top_srcdir) $(LIBEE_CFLAGS) $(LIBESTR_CFLAGS)
-normalizer_LDADD = $(LIBEE_LIBS) $(LIBLOGNORM_LIBS) $(LIBESTR_LIBS)
+normalizer_LDADD = $(LIBEE_LIBS) liblognorm.la $(LIBESTR_LIBS)
 
 lib_LTLIBRARIES = liblognorm.la
 
