use host to resolve domain
--- gajim-0.12.1/src/common/nslookup.py.orig	2009-04-18 02:11:36.000000000 +0800
+++ gajim-0.12.1/src/common/nslookup.py	2009-04-18 02:14:37.000000000 +0800
@@ -123,11 +123,11 @@
 				line = helpers.decode_string(line)
 				domain = ufqdn
 			if domain:
-				rest = line[len(domain):].split('=')
+				rest = line[len(domain):].split('record')
 				if len(rest) != 2:
 					continue
 				answer_type, props_str = rest
-				if answer_type.strip() != 'service':
+				if answer_type.strip() != 'has SRV':
 					continue
 				props = props_str.strip().split(' ')
 				if len(props) < 4:
@@ -296,7 +296,7 @@
 			return
 	
 	def _compose_command_args(self):
-		return ['nslookup', '-type=' + self.type , self.host]
+		return ['host', '-t ' + self.type , self.host]
 	
 	def _return_result(self):
 		if self.result_handler:
