diff -upNr nvclock-0.8b4/src/Makefile.in.orig nvclock-0.8b4/src/Makefile.in
--- nvclock-0.8b4/src/Makefile.in.orig	Sat Jan  3 12:02:54 2009
+++ nvclock-0.8b4/src/Makefile.in	Thu May  6 08:40:39 2010
@@ -50,8 +50,8 @@
 install: install-recursive install-app
 install-app:
 	mkdir -p $(bindir)
-	$(INSTALL) -c nvclock $(DESTDIR)$(bindir)/nvclock
-	$(INSTALL) -c smartdimmer $(DESTDIR)$(bindir)/smartdimmer
+	$(INSTALL) -c nvclock $(bindir)/nvclock
+	$(INSTALL) -c smartdimmer $(bindir)/smartdimmer
 
 
 uninstall: uninstall-recursive uninstall-app
