--- memtest86+-5.01/controller.c	2013-08-10 02:01:58.000000000 +0000
+++ memtest86+-5.01/controller.c	2013-12-12 20:58:12.873555378 +0000
@@ -292,7 +292,7 @@
 
 	/* First, locate the PCI bus where the MCH is located */
 
-	for(i = 0; i < sizeof(possible_nhm_bus); i++) {
+	for(i = 0; i < sizeof(possible_nhm_bus) / sizeof(possible_nhm_bus[0]); i++) {
 		pci_conf_read( possible_nhm_bus[i], 3, 4, 0x00, 2, &vid);
 		pci_conf_read( possible_nhm_bus[i], 3, 4, 0x02, 2, &did);
 		vid &= 0xFFFF;
@@ -327,7 +327,7 @@
 	ctrl.mode = ECC_NONE;
 
 	/* First, locate the PCI bus where the MCH is located */
-	for(i = 0; i < sizeof(possible_nhm_bus); i++) {
+	for(i = 0; i < sizeof(possible_nhm_bus) / sizeof(possible_nhm_bus[0]); i++) {
 		pci_conf_read( possible_nhm_bus[i], 3, 4, 0x00, 2, &vid);
 		pci_conf_read( possible_nhm_bus[i], 3, 4, 0x02, 2, &did);
 		vid &= 0xFFFF;
@@ -1424,7 +1424,7 @@
 	unsigned long dramchr;
 	float clockratio;
 	double dramclock;
-	unsigned int dummy[3];
+	unsigned int dummy[4];
 	int ram_type;
 
 	float coef = 10;
@@ -3680,7 +3680,7 @@
 static void poll_timings_k16(void) {
 
 	ulong dramt0, dramt1;
-	int cas, rcd, rp, rc, ras;
+	int cas, rcd, rp, ras;
 	
 	pci_conf_read(0, 24, 2, 0x200, 4, &dramt0);
 	pci_conf_read(0, 24, 2, 0x204, 4, &dramt1);	
@@ -3690,7 +3690,7 @@
 	rp = ((dramt0 >> 16) & 0x1F);
 	ras = ((dramt0 >> 24) & 0x3F);
 	
-	rc = (dramt1 & 0x3F);	
+	//int rc = (dramt1 & 0x3F);	
 	
 	print_ram_line(cas, rcd, rp, ras, 1);
 }
