--- memtest86+-5.01/setup.S
+++ memtest86+-5.01/setup.S
@@ -27,11 +27,11 @@
 #ljmp	$INITSEG, $(reload - start + 0x200)
 reload:
 	movw	$INITSEG, %ax
-	movw	%ax, %ds
 	movw	%ax, %es
 	movw	%ax, %fs
+	movw	%ax, %gs
 	movw	%ax, %ss	# reset the stack to INITSEG:0x4000-12.
-	movw	%dx, %sp
+	movw	$0x4000-12, %sp
 	push	%cs
 	pop	%ds
 	lidt	idt_48 - start	# load idt with 0,0
@@ -50,13 +50,13 @@
      	jz	alt_a20_done
      	
      	/* set or clear bit1, the ALT_A20_GATE bit */
-     	movb	4(%esp), %ah
-     	testb	%ah, %ah
-     	jz	alt_a20_cont1
+     	//movb	4(%esp), %ah
+     	//testb	%ah, %ah
+     	//jz	alt_a20_cont1
      	orb	$2, %al
-     	jmp	alt_a20_cont2
+     	//jmp	alt_a20_cont2
 alt_a20_cont1:
-	and	$0xfd, %al
+	//and	$0xfd, %al
 
 	/* clear the INIT_NOW bit; don't accidently reset the machine */
 alt_a20_cont2:
@@ -87,9 +87,9 @@
 	movw	$KERNEL_DS, %ax
 	movw	%ax, %ds
 	movw	%ax, %es
-	movw	%ax, %ss
 	movw	%ax, %fs
 	movw	%ax, %gs
+	movw	%ax, %ss
 
 data32	ljmp	$KERNEL_CS, $(TSTLOAD <<4)	# jmp offset 2000 of segment 0x10 (cs)
 
@@ -100,18 +100,16 @@
  * No timeout is used - if this hangs there is something wrong with
  * the machine, and we probably couldn't proceed anyway.
  */
+has_output:
+	call	delay
+	inb	$0x60, %al	# read it
 empty_8042:
 	call	delay
 	inb	$0x64, %al	# 8042 status port
 	cmpb	$0xff, %al	# from grub-a20-patch, skip if not impl
 	jz	empty_8042_ret
 	testb	$1, %al		# output buffer?
-	jz	no_output
-	call	delay
-	inb	$0x60, %al	# read it
-	jmp	empty_8042
-	
-no_output:
+	jnz	has_output
 	testb	$2, %al		# is input buffer full?
 	jnz	empty_8042	# yes - loop
 empty_8042_ret:
