--- etc/nanorc.orig	2019-05-03 14:19:11.780800756 +0100
+++ etc/nanorc	2019-05-03 14:32:51.179322558 +0100
@@ -1,7 +1,6 @@
 ## Sample initialization file for GNU nano.
-##
-## This file should not be in DOS or Mac format, and characters
-## specially interpreted by the shell should not be escaped here.
+## /etc/nanorc: system-wide initialization file for nano.
+## ~/.nanorc:   personal config file.
 ##
 ## To make sure an option is disabled, use "unset <option>".
 ##
@@ -62,7 +61,7 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Scroll the buffer contents per half-screen instead of per line.
 # set jumpyscrolling
@@ -84,7 +83,7 @@
 ## mouse clicks can be used to place the cursor, set the mark (with a
 ## double click), and execute shortcuts.  The mouse will work in the X
 ## Window System, and on the console when gpm is running.
-# set mouse
+set mouse
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
@@ -161,7 +160,7 @@
 # set suspend
 
 ## Use this tab size instead of the default; it must be greater than 0.
-# set tabsize 8
+set tabsize 4
 
 ## Convert typed tabs to spaces.
 # set tabstospaces
@@ -305,3 +304,151 @@
 #unbind M-J main
 #unbind M-T main
 #set multibuffer
+
+
+
+## Assembler
+# include "/usr/share/nano/asm.nanorc"
+
+## Autoconf
+# include "/usr/share/nano/autoconf.nanorc"
+
+## AWK
+# include "/usr/share/nano/awk.nanorc"
+
+## C/C++
+include "/usr/share/nano/c.nanorc"
+
+## Changelogs
+# include "/usr/share/nano/changelog.nanorc"
+
+## CMake
+# include "/usr/share/nano/cmake.nanorc"
+
+## CSS
+include "/usr/share/nano/css.nanorc"
+
+## Apt's sources.list
+# include "/usr/share/nano/debian.nanorc"
+
+## Default
+include "/usr/share/nano/default.nanorc"
+
+## Emacs Lisp
+# include "/usr/share/nano/elisp.nanorc"
+
+## Fortran 90/95
+# include "/usr/share/nano/fortran.nanorc"
+
+## Gentoo ebuilds/eclasses
+# include "/usr/share/nano/gentoo.nanorc"
+
+## Go
+# include "/usr/share/nano/go.nanorc"
+
+## Groff
+# include "/usr/share/nano/groff.nanorc"
+
+## Guile Scheme
+# include "/usr/share/nano/guile.nanorc"
+
+## HTML
+# include "/usr/share/nano/html.nanorc"
+
+## Java
+# include "/usr/share/nano/java.nanorc"
+
+## Javascript
+include "/usr/share/nano/javascript.nanorc"
+
+## JSON
+# include "/usr/share/nano/json.nanorc"
+
+## Lua
+# include "/usr/share/nano/lua.nanorc"
+
+## Makefiles
+include "/usr/share/nano/makefile.nanorc"
+
+## Manpages
+include "/usr/share/nano/man.nanorc"
+
+## Magicpoint presentations
+# include "/usr/share/nano/mgp.nanorc"
+
+## Quoted emails (under e.g. mutt)
+include "/usr/share/nano/mutt.nanorc"
+
+## Nanorc files
+include "/usr/share/nano/nanorc.nanorc"
+
+## nftables
+# include "/usr/share/nano/nftables.nanorc"
+
+## C/C++/Obj-C
+# include "/usr/share/nano/objc.nanorc"
+
+## OCaml
+# include "/usr/share/nano/ocaml.nanorc"
+
+## Patch files
+include "/usr/share/nano/patch.nanorc"
+
+## Perl
+# include "/usr/share/nano/perl.nanorc"
+
+## PHP
+include "/usr/share/nano/php.nanorc"
+
+## PO files
+include "/usr/share/nano/po.nanorc"
+
+## PostgreSQL
+# include "/usr/share/nano/postgresql.nanorc"
+
+## POV-Ray
+# include "/usr/share/nano/pov.nanorc"
+
+## Python
+include "/usr/share/nano/python.nanorc"
+
+## Ruby
+# include "/usr/share/nano/ruby.nanorc"
+
+## Bourne shell scripts
+include "/usr/share/nano/sh.nanorc"
+
+## RPM spec files
+# include "/usr/share/nano/spec.nanorc"
+
+## Tcl files
+# include "/usr/share/nano/tcl.nanorc"
+
+## TeX files
+# include "/usr/share/nano/tex.nanorc"
+
+## Texinfo files
+# include "/usr/share/nano/texinfo.nanorc"
+
+## XML files
+include "/usr/share/nano/xml.nanorc"
+
+
+## Config file - By SliTaz
+syntax "conf" "\.?conf$"
+## Keywords
+icolor brightgreen "="
+## Comments
+icolor cyan "^[[:space:]]*#.*$"
+
+## HTML - With modification for SliTaz.
+syntax "HTML" "\.html$"
+color brightblue start="<" end=">"
+color red "&[^; 	]*;"
+color brightwhite start="<!DOCTYPE" end=">" start="<title>" end="</title>" start="<!--" end="-->"
+color brightcyan start="<meta" end=">" start="<link" end=">"
+color brightyellow "<p>|</p>|<pre>|</pre>|<li>|</li>"
+color yellow "<ul>|</ul>|<hr>|<hr />"
+color brightred "href=|name=|rel=|http-equiv=|content=|title=|type|style"
+color brightgreen "<h1>|</h1>|<h2>|</h2>|<h3>|</h3>|<h4>|</h4>|<h5>|</h5>|<h6>|</h6>"
+
