--- miniperlmain.c.orig
+++ miniperlmain.c
@@ -117,13 +117,6 @@
     if (!exitstatus)
         perl_run(my_perl);
 
-    /* Unregister our signal handler before destroying my_perl */
-    for (i = 0; PL_sig_name[i]; i++) {
-	if (rsignal_state(PL_sig_num[i]) == (Sighandler_t) PL_csighandlerp) {
-	    rsignal(PL_sig_num[i], (Sighandler_t) SIG_DFL);
-	}
-    }
-
     exitstatus = perl_destruct(my_perl);
 
     perl_free(my_perl);
