--- a/Makefile.in
+++ b/Makefile.in
@@ -729,11 +729,11 @@
 @INTLTOOL_SCHEMAS_RULE@
 
 @GCONF_SCHEMAS_INSTALL_TRUE@install-data-local:
-@GCONF_SCHEMAS_INSTALL_TRUE@        if test -z "$(DESTDIR)" ; then \
-@GCONF_SCHEMAS_INSTALL_TRUE@          for p in $(schemas_DATA) ; do \
-@GCONF_SCHEMAS_INSTALL_TRUE@            GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
-@GCONF_SCHEMAS_INSTALL_TRUE@          done \
-@GCONF_SCHEMAS_INSTALL_TRUE@        fi
+@GCONF_SCHEMAS_INSTALL_TRUE@	if test -z "$(DESTDIR)" ; then \
+@GCONF_SCHEMAS_INSTALL_TRUE@	  for p in $(schemas_DATA) ; do \
+@GCONF_SCHEMAS_INSTALL_TRUE@	    GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
+@GCONF_SCHEMAS_INSTALL_TRUE@	  done \
+@GCONF_SCHEMAS_INSTALL_TRUE@	fi
 @GCONF_SCHEMAS_INSTALL_FALSE@install-data-local:
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- a/libgksu/Makefile.in
+++ b/libgksu/Makefile.in
@@ -267,7 +267,7 @@
 gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
 gksu_run_helper_SOURCES = gksu-run-helper.c
 test_gksu_SOURCES = test-gksu.c
-test_gksu_LDADD = libgksu2.la
+test_gksu_LDADD = libgksu2.la -lgtk-x11-2.0
 test_gksu_LDFLAGS = `pkg-config --libs glib-2.0`
 EXTRA_DIST = libgksu.ver
 all: all-am
--- a/libgksuui/Makefile.in
+++ b/libgksuui/Makefile.in
@@ -231,7 +231,7 @@
 libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 test_gksuui_SOURCES = test-gksuui.c
-test_gksuui_LDADD = libgksuui1.0.la
+test_gksuui_LDADD = libgksuui1.0.la -lX11
 test_gksuui_LDFLAGS = `pkg-config --libs glib-2.0`
 all: all-am
 
--- a/gksu-properties/gksu-properties.desktop.in
+++ b/gksu-properties/gksu-properties.desktop.in
@@ -3,9 +3,9 @@
 _Name=Privilege granting
 _Comment=Configure behavior of the privilege-granting tool
 Exec=gksu-properties
-Icon=gksu.png
+Icon=gksu
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Application;Settings;AdvancedSettings;
+Categories=GNOME;GTK;Settings;X-AdvancedSettings;
-OnlyShowIn=GNOME;XFCE;
+OnlyShowIn=GNOME;XFCE;LXDE;
