--- portmap_5beta/Makefile	Fri Feb  1 11:54:29 2008
+++ Makefile	Fri Feb  1 11:55:19 2008
@@ -15,9 +15,9 @@
 # no access control tables. The local system, since it runs the portmap
 # daemon, is always treated as an authorized host.
 
-HOSTS_ACCESS= -DHOSTS_ACCESS
+#HOSTS_ACCESS= -DHOSTS_ACCESS
 #WRAP_LIB = $(WRAP_DIR)/libwrap.a
-WRAP_LIB = -lwrap
+#WRAP_LIB = -lwrap
 
 # Comment out if your RPC library does not allocate privileged ports for
 # requests from processes with root privilege, or the new portmap will
@@ -25,7 +25,7 @@
 # ports. You can find out by running "rpcinfo -p"; if all mountd and NIS
 # daemons use a port >= 1024 you should probably disable the next line.
 
-CHECK_PORT = -DCHECK_PORT
+#CHECK_PORT = -DCHECK_PORT
 
 # Warning: troublesome feature ahead!! Enable only when you are really
 # desperate!!
