--- src/celengine/overlay.h	2010-04-16 17:43:35.000000000 +0200
+++ src/celengine/overlay.h	2010-04-16 17:43:55.000000000 +0200
@@ -12,6 +12,7 @@
 
 #include <string>
 #include <iostream>
+#include <cstdio>
 #include <celtxf/texturefont.h>
 
 
--- src/celmath/mathlib.h	2010-04-16 18:16:09.000000000 +0200
+++ src/celmath/mathlib.h	2010-04-16 18:16:18.000000000 +0200
@@ -44,11 +44,6 @@ template<class T> T radToDeg(T r)
     return r * 180 / static_cast<T>(PI);
 }
 
-template<class T> T abs(T x)
-{
-    return (x < 0) ? -x : x;
-}
-
 template<class T> T square(T x)
 {
     return x * x;
--- src/celengine/star.cpp	2010-04-16 18:17:42.000000000 +0200
+++ src/celengine/star.cpp	2010-04-16 18:17:57.000000000 +0200
@@ -10,6 +10,7 @@
 #include <celmath/mathlib.h>
 #include <cstring>
 #include <cassert>
+#include <cstdio>
 #include "celestia.h"
 #include "astro.h"
 #include "orbit.h"
--- src/celestia/celx_object.cpp	2010-04-16 18:19:52.000000000 +0200
+++ src/celestia/celx_object.cpp	2010-04-16 18:22:37.000000000 +0200
@@ -773,7 +773,7 @@ static int object_mark(lua_State* l)
         markAlpha = 1.0f;
     
     Color markColorAlpha(0.0f, 1.0f, 0.0f, 0.9f);
-    markColorAlpha = Color::Color(markColor, markAlpha);
+    markColorAlpha = Color(markColor, markAlpha);
     
     const char* markLabel = celx.safeGetString(6, WrongType, "Fifth argument to object:mark must be a string");
     if (markLabel == NULL)
