--- a/Makefile.in
+++ b/Makefile.in
@@ -14,8 +14,8 @@
 BINDIR = $(prefix)/bin
 LIBDIR = $(prefix)/lib
 INCDIR = $(prefix)/include
-MAN1DIR = $(prefix)/man/man1
-MAN3DIR = $(prefix)/man/man3
+MAN1DIR = $(prefix)/share/man/man1
+MAN3DIR = $(prefix)/share/man/man3
 INFODIR = $(prefix)/info
 
 # getopt may be installed or not, if not take our copy
@@ -63,14 +63,14 @@
 	cd doc && $(MAKE) 
 
 install:
-	$(INSTALL) -d $(BINDIR) $(INCDIR) $(LIBDIR) $(MAN1DIR) \
-		$(MAN3DIR) $(INFODIR)
-	$(INSTALL) -c $(TARGET) $(BINDIR)
-	$(INSTALL) -c -m 0644 $(HEADER) $(INCDIR)
-	$(INSTALL) -c -m 0644  $(LIBRARY) $(LIBDIR)
-	$(INSTALL) -c -m 0644  $(MAN1) $(MAN1DIR)
-	$(INSTALL) -c -m 0644  $(MAN3) $(MAN3DIR)
-	$(INSTALL) -c -m 0644  $(INFO) $(INFODIR)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR) $(DESTDIR)$(INCDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(MAN1DIR) \
+		$(DESTDIR)$(MAN3DIR) $(DESTDIR)$(INFODIR)
+	$(INSTALL) -c $(TARGET) $(DESTDIR)$(BINDIR)
+	$(INSTALL) -c -m 0644 $(HEADER) $(DESTDIR)$(INCDIR)
+	$(INSTALL) -c -m 0644  $(LIBRARY) $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -c -m 0644  $(MAN1) $(DESTDIR)$(MAN1DIR)
+	$(INSTALL) -c -m 0644  $(MAN3) $(DESTDIR)$(MAN3DIR)
+	$(INSTALL) -c -m 0644  $(INFO) $(DESTDIR)$(INFODIR)
 
 uninstall:
 	$(RM) -f $(BINDIR)/$(TARGET)
