--- dvd+rw-tools-7.1/growisofs.c	Tue Mar  4 09:15:03 2008
+++ dvd+rw-tools-7.1-new/growisofs.c	Wed Apr 22 21:57:43 2009
@@ -442,6 +442,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <assert.h>
+#include <limits.h>
 #include "mp.h"
 
 #if defined(__unix) || defined(__unix__)
--- dvd+rw-tools-7.1/transport.hxx	Sat Mar  1 10:34:43 2008
+++ dvd+rw-tools-7.1-new/transport.hxx	Wed Apr 22 21:58:09 2009
@@ -145,6 +145,7 @@
 #include <sys/wait.h>
 #include <sys/utsname.h>
 #include <scsi/sg.h>
+#include <limits.h>
 #if !defined(SG_FLAG_LUN_INHIBIT)
 # if defined(SG_FLAG_UNUSED_LUN_INHIBIT)
 #  define SG_FLAG_LUN_INHIBIT SG_FLAG_UNUSED_LUN_INHIBIT
