--- src/plugins/cpu/cpu.c
+++ src/plugins/cpu/cpu.c.taz
@@ -229,7 +229,7 @@
     /* Clone a graphics context and set "green" as its foreground color.
      * We will use this to draw the graph. */
     c->graphics_context = gdk_gc_new(p->panel->topgwin->window);
-    gdk_color_parse("green",  &c->foreground_color);
+    gdk_color_parse("#cfcfcf",  &c->foreground_color);
     gdk_colormap_alloc_color(gdk_drawable_get_colormap(p->panel->topgwin->window), &c->foreground_color, FALSE, TRUE);
     gdk_gc_set_foreground(c->graphics_context, &c->foreground_color);
 
