--- src/plugins/cpu/cpu.c
+++ src/plugins/cpu/cpu.c.taz
@@ -222,14 +222,14 @@
 
     /* Allocate drawing area as a child of top level widget.  Enable button press events. */
     c->da = gtk_drawing_area_new();
-    gtk_widget_set_size_request(c->da, 40, PANEL_HEIGHT_DEFAULT);
+    gtk_widget_set_size_request(c->da, 26, PANEL_HEIGHT_DEFAULT);
     gtk_widget_add_events(c->da, GDK_BUTTON_PRESS_MASK);
     gtk_container_add(GTK_CONTAINER(p->pwid), c->da);
 
     /* Clone a graphics context and set "green" as its foreground color.
      * We will use this to draw the graph. */
     c->graphics_context = gdk_gc_new(p->panel->topgwin->window);
-    gdk_color_parse("green",  &c->foreground_color);
+    gdk_color_parse("#cfcfcf",  &c->foreground_color);
     gdk_colormap_alloc_color(gdk_drawable_get_colormap(p->panel->topgwin->window), &c->foreground_color, FALSE, TRUE);
     gdk_gc_set_foreground(c->graphics_context, &c->foreground_color);
 
