--- resources.qrc
+++ resources.qrc
@@ -8,5 +8,18 @@
         <file>images/item/track.png</file>
         <file>style.css</file>
         <file>images/sort.png</file>
+        <file>images/audio-volume-high.png</file>
+        <file>images/audio-volume-muted.png</file>
+        <file>images/edit-clear.png</file>
+        <file>images/go-previous.png</file>
+        <file>images/gtk-info.png</file>
+        <file>images/list-add.png</file>
+        <file>images/media-playback-start.png</file>
+        <file>images/media-playback-stop.png</file>
+        <file>images/media-playlist-repeat.png</file>
+        <file>images/media-playlist-shuffle.png</file>
+        <file>images/media-skip-backward.png</file>
+        <file>images/media-skip-forward.png</file>
+        <file>images/view-restore.png</file>
     </qresource>
 </RCC>
