--- computer.h	2008-05-02 13:32:10.000000000 +0200
+++ computer.h	2008-05-02 13:34:31.000000000 +0200
@@ -23,6 +23,7 @@
 static struct {
     gchar *file, *codename;
 } distro_db[] = {
+    { DB_PREFIX "slitaz-release",	"slitaz" },
     { DB_PREFIX "debian_version",	"deb"  },
     { DB_PREFIX "slackware-version",	"slk"  },
     { DB_PREFIX "mandrake-release",	"mdk"  },
--- arch/linux/common/os.h	2008-05-02 13:43:33.000000000 +0200
+++ arch/linux/common/os.h	2008-05-02 13:44:49.000000000 +0200
@@ -215,6 +215,10 @@
 		    ((buf[0] >= '0' && buf[0] <= '9') || buf[0] != 'D')) {
 		    os->distro = g_strdup_printf
 			("Debian GNU/Linux %s", buf);
+		}
+		else if (!strncmp(distro_db[i].codename, "slitaz", 6)) {
+		    os->distro = g_strdup_printf
+			("Slitaz GNU/Linux %s", buf);
 		} else {
 		    os->distro = g_strdup(buf);
 		}
