--- arch/common/users.h	Sun Nov  4 14:14:35 2007
+++ arch/common/users.h	Sun Apr 20 23:57:31 2008
@@ -44,7 +44,7 @@
 
       uid = atoi(tmp[2]);
       strend(tmp[4], ',');
-      if (uid >= 1000 && uid <= 65530) {
+      if (uid >= 500 && uid <= 65530) {
         human_users = h_strdup_cprintf("$%s$%s=%s\n", human_users, key, tmp[0], tmp[4]);
       } else {
         sys_users = h_strdup_cprintf("$%s$%s=%s\n", sys_users, key, tmp[0], tmp[4]);
