--- spacefm-0.8.4.orig/src/vfs/vfs-file-task.c
+++ spacefm-0.8.4/src/vfs/vfs-file-task.c
@@ -1725,6 +1725,12 @@
             argv[a++] = g_strdup( "SpaceFM Command" );
             single_arg = TRUE;
         }
+        else if ( !strcmp( use_su, "/usr/bin/tazbox" ) )
+        {
+            // tazbox
+            argv[a++] = g_strdup( "su" );
+            single_arg = TRUE;
+        }
         else if ( strstr( use_su, "kdesu" ) )
         {
             // kdesu kdesudo
