--- a/configure
+++ b/configure
@@ -1660,13 +1660,13 @@
   initial_path="\$4"
   files="\$5"
   act="\${ITCHCONFIG_ACT-}"
-  ic_makedir="\${MKDIRPROG-mkdir}"
+  ic_makedir="mkdir -p"
   ic_program_prefix="$ic_program_prefix"
   ic_program_suffix="$ic_program_suffix"
   checked_path=
   for one_file in \$files
   do
-    path="\$initial_path"
+    path="\$DESTDIR\$initial_path"
     file=\`echo "\$one_file" | sed 's,.*[/],,'\`
     case \$magic in
     m0) ;;
