--- a/docs/reference/lib/Makefile.in
+++ b/docs/reference/lib/Makefile.in
@@ -713,8 +713,6 @@
 	  if test -n "$(DOC_MODULE_VERSION)"; then \
 	    mv -f $${installdir}/$(DOC_MODULE).devhelp2 \
 	      $${installdir}/$(DOC_MODULE)-$(DOC_MODULE_VERSION).devhelp2; \
-	    mv -f $${installdir}/$(DOC_MODULE).devhelp \
-	      $${installdir}/$(DOC_MODULE)-$(DOC_MODULE_VERSION).devhelp; \
 	  fi; \
 	  $(GTKDOC_REBASE) --relative --dest-dir=$(DESTDIR) --html-dir=$${installdir}; \
 	fi

--- a/docs/reference/ui/Makefile.in
+++ b/docs/reference/ui/Makefile.in
@@ -724,8 +724,6 @@
 @WITH_GTK_UI_TRUE@	  if test -n "$(DOC_MODULE_VERSION)"; then \
 @WITH_GTK_UI_TRUE@	    mv -f $${installdir}/$(DOC_MODULE).devhelp2 \
 @WITH_GTK_UI_TRUE@	      $${installdir}/$(DOC_MODULE)-$(DOC_MODULE_VERSION).devhelp2; \
-@WITH_GTK_UI_TRUE@	    mv -f $${installdir}/$(DOC_MODULE).devhelp \
-@WITH_GTK_UI_TRUE@	      $${installdir}/$(DOC_MODULE)-$(DOC_MODULE_VERSION).devhelp; \
 @WITH_GTK_UI_TRUE@	  fi; \
 @WITH_GTK_UI_TRUE@	  $(GTKDOC_REBASE) --relative --dest-dir=$(DESTDIR) --html-dir=$${installdir}; \
 @WITH_GTK_UI_TRUE@	fi

--- a/po/LINGUAS
+++ b/po/LINGUAS
@@ -1,3 +1,4 @@
+ru
 zh_CN
 zh_HK
 zh_TW

--- a/data/RhythmCat2.desktop
+++ b/data/RhythmCat2.desktop
@@ -8,7 +8,7 @@
 Exec=RhythmCat2 %U
 StartupNotify=false
 Type=Application
-Name[en_US]=RhythmCat2 Music Player
-Name[zh_CN]=RhythmCat2 音乐播放器
-Name[zh_TW]=RhythmCat2 音樂播放器
+Comment[ru]=Музыкальный проигрыватель RhythmCat2
+Comment[zh_CN]=RhythmCat2 音乐播放器
+Comment[zh_TW]=RhythmCat2 音樂播放器
 MimeType=application/x-ogg;application/ogg;audio/x-vorbis+ogg;audio/x-scpls;audio/x-mp3;audio/x-mpeg;audio/mpeg;audio/x-mpegurl;audio/x-flac;

--- a/ui/rc-ui-plugin.c
+++ b/ui/rc-ui-plugin.c
@@ -155,7 +155,7 @@
     g_object_set(about_dialog, "program-name", plugin_data->info->name,
         "authors", authors, "comments", plugin_data->info->description,
         "website", plugin_data->info->homepage, "version",
-        plugin_data->info->version, NULL);
+        plugin_data->info->version, "logo", rc_ui_player_get_icon_image(), NULL);
     gtk_dialog_run(GTK_DIALOG(about_dialog));
     gtk_widget_destroy(about_dialog);
 }
