--- daemon/gvfsbackendsftp.c.orig
+++ daemon/gvfsbackendsftp.c
@@ -237,7 +237,8 @@
       if (ssh_stderr == NULL)
 	res = SFTP_VENDOR_INVALID;
       else if ((strstr (ssh_stderr, "OpenSSH") != NULL) ||
-	       (strstr (ssh_stderr, "Sun_SSH") != NULL))
+	       (strstr (ssh_stderr, "Sun_SSH") != NULL) ||
+	       (strstr (ssh_stderr, "Dropbear") != NULL))
 	res = SFTP_VENDOR_OPENSSH;
       else if (strstr (ssh_stderr, "SSH Secure Shell") != NULL)
 	res = SFTP_VENDOR_SSH;
