diff -Naur ncurses-ruby-1.3.1/form_wrap.c ncurses-ruby-1.3.1.format/form_wrap.c
--- ncurses-ruby-1.3.1/form_wrap.c	2014-02-03 11:31:39.049867878 +0100
+++ ncurses-ruby-1.3.1.format/form_wrap.c	2014-02-03 11:44:37.001008654 +0100
@@ -1132,7 +1132,7 @@
 		  char msg[500];
 		  snprintf(msg, 500, "The validation functions for this field type need %d additional arguments.",(int)(NUM2INT(argc)-1));
 		  msg[499]=0;
-		  rb_raise(rb_eArgError, msg);	
+		  rb_raise(rb_eArgError, "%s", msg);	
 		}
 	 } 
   }
