--- src/base/fm-mime-type.c.orig
+++ src/base/fm-mime-type.c
@@ -156,10 +156,10 @@
         if(uncertain)
         {
             int fd, len;
-            if(pstat->st_size == 0) /* empty file = text file with 0 characters in it. */
+            if(pstat->st_size == 0) /* empty file */
             {
                 g_free(type);
-                return fm_mime_type_from_name("text/plain");
+                return fm_mime_type_from_name("application/x-zerosize");
             }
             fd = open(file_path, O_RDONLY);
             if(fd >= 0)
