--- a/src/DataColumns.h
+++ b/src/DataColumns.h
@@ -186,39 +186,12 @@
 	 **/
 	DataColumns();
 
-
 	// Data members
 
 	static DataColumns * _instance;
 	DataColumnList	     _columns;
 
     };	// class DataColumns
-
-
-    /**
-     * Print a DataColumn in text form to a debug stream.
-     **/
-    inline QTextStream & operator<< ( QTextStream & stream, DataColumn col )
-    {
-	stream << DataColumns::toString( col );
-
-	return stream;
-    }
-
-
-    /**
-     * Print a DataColumn in text form to a debug stream.
-     **/
-    inline QTextStream & operator<< ( QTextStream &          stream,
-                                      const DataColumnList & colList )
-    {
-	stream << "[ "
-               << DataColumns::toStringList( colList ).join( ", " )
-               << " ]";
-
-	return stream;
-    }
-
 
 }	// namespace QDirStat

--- a/src/DataColumns.cpp
+++ b/src/DataColumns.cpp
@@ -13,8 +13,31 @@
 #include "Logger.h"
 #include "Exception.h"
 
 using namespace QDirStat;
 
+    /**
+     * Print a DataColumn in text form to a debug stream.
+     **/
+    inline QTextStream & operator<< ( QTextStream & stream, DataColumn col )
+    {
+	stream << DataColumns::toString( col );
+
+	return stream;
+    }
+
+
+    /**
+     * Print a DataColumn in text form to a debug stream.
+     **/
+    inline QTextStream & operator<< ( QTextStream &          stream,
+                                      const DataColumnList & colList )
+    {
+	stream << "[ "
+               << DataColumns::toStringList( colList ) //.join( ", " )
+               << " ]";
+ 
+	return stream;
+    }
 
 DataColumns * DataColumns::_instance = 0;

--- a/man/man.pro
+++ b/man/man.pro
@@ -7,13 +7,13 @@
 
 MAN_SRC      = $$files( *.1 )
 MAN_TARGET   = $$MAN_SRC
-MAN_TARGET  ~= s/.1$/.1.gz/g
+#MAN_TARGET  ~= s/.1$/.1.gz/g
 
 # message(src:    $$MAN_SRC)
 # message(target: $$MAN_TARGET)
 
 man.files    = $$MAN_TARGET
-man.commands = gzip --keep --force $$MAN_SRC
+#man.commands = gzip --keep --force $$MAN_SRC
 man.path     = $$INSTALL_PREFIX/share/man/man1
 
 INSTALLS    += man
