--- xarchive-0.2.8-6.ori/wrappers/tar-gnu.src	Tue Nov  8 20:50:59 2005
+++ xarchive-0.2.8-6/wrappers/tar-gnu.src	Tue Mar 11 23:28:34 2008
@@ -1,4 +1,4 @@
-#! /path/to/bash
+#! /bin/bash
 # tar-wrap.sh - bash tar wrapper for xarchive frontend
 # Copyright (C) 2005 Lee Bigelow <ligelowbee@yahoo.com> 
 # 
@@ -54,14 +54,14 @@
     if [ $(expr "$lc_archive" : ".*\."$ext"$") -gt 0 ]; then
         DECOMPRESS="gzip -dc"
         COMPRESS="gzip -c"
-        TAR_COMPRESS_OPT="--use-compress-prog=gzip"
+        TAR_COMPRESS_OPT="z"
     fi
 done
 for ext in $BZIP2_EXTS; do
     if [ $(expr "$lc_archive" : ".*\."$ext"$") -gt 0 ]; then
         DECOMPRESS="bzip2 -dc" 
         COMPRESS="bzip2 -c"
-        TAR_COMPRESS_OPT="--use-compress-prog=bzip2"
+        TAR_COMPRESS_OPT="j"
     fi
 done
 for ext in $COMPRESS_EXTS; do
@@ -76,8 +76,8 @@
 # open and extract can use tar's cmd line option
 # add, new, and remove need to decompress the tar first
 # do their thing, than recompress the tar.
-OPEN_OPTS="$TAR_COMPRESS_OPT -tvf"
-EXTRACT_OPTS="$TAR_COMPRESS_OPT -xf"
+OPEN_OPTS="tv${TAR_COMPRESS_OPT}f"
+EXTRACT_OPTS="-x${TAR_COMPRESS_OPT}f"
 ADD_OPTS="-rf"
 NEW_OPTS="-cf"
 REMOVE_OPTS="--delete -f"
