# opened ticket at https://bugs.launchpad.net/marlin/+bug/1404153
--- marlin-0.5/libwidgets/tests/tests-pathbar.vala
+++ marlin-0.5/libwidgets/tests/tests-pathbar.vala
@@ -35,7 +35,7 @@
         var bread_entry = new BreadcrumbsEntry();
         assert(bread_entry is BreadcrumbsEntry);
         assert(bread_entry.text == "");
-        Gdk.EventKey event = Gdk.EventKey();
+        Gdk.EventKey event = new Gdk.EventKey();
         event.window = breads.get_window();
         event.keyval = 0x061; /* a */
         bread_entry.key_press_event(event);
@@ -122,7 +122,7 @@
         assert(bread_entry.text == "");
         bread_entry.text = "abdcefghij/";
         bread_entry.cursor = ("abdcefghij/").length;
-        Gdk.EventKey event = Gdk.EventKey();
+        Gdk.EventKey event = new Gdk.EventKey();
         event.state = Gdk.ModifierType.SHIFT_MASK;
         event.window = breads.get_window();
         event.keyval = 0xff51; /* left */
@@ -136,7 +136,7 @@
         var bread_entry = new BreadcrumbsEntry();
         assert(bread_entry is BreadcrumbsEntry);
         assert(bread_entry.text == "");
-        Gdk.EventKey event = Gdk.EventKey();
+        Gdk.EventKey event = new Gdk.EventKey();
         event.window = breads.get_window();
         event.keyval = 0xff08; /* backspace */
         bread_entry.key_press_event(event);
@@ -149,7 +149,7 @@
         BreadcrumbsEntry bread_entry = breads.entry;
         assert(bread_entry is BreadcrumbsEntry);
         assert(bread_entry.text == "");
-        Gdk.EventKey event = Gdk.EventKey();
+        Gdk.EventKey event = new Gdk.EventKey();
         event.window = breads.get_window();
         event.keyval = 0xff08; /* backspace */
         breads.key_press_event(event);
