--- alsa-utils-1.0.18.ori/alsactl/Makefile.am	Wed Oct 29 13:42:11 2008
+++ alsa-utils-1.0.18/alsactl/Makefile.am	Wed Dec 10 22:47:15 2008
@@ -1,11 +1,11 @@
 SUBDIRS = init
 
 sbin_PROGRAMS=alsactl
-man_MANS=alsactl.1 alsactl_init.7
+#man_MANS=alsactl.1 alsactl_init.7
 EXTRA_DIST=alsactl.1 alsactl_init.xml
 
 alsactl_SOURCES=alsactl.c state.c utils.c init_parse.c
 noinst_HEADERS=alsactl.h list.h init_sysdeps.c init_utils_string.c init_utils_run.c init_sysfs.c
 
-%.7: %.xml
-	xmlto man $?
+#%.7: %.xml
+#	xmlto man $?
--- alsa-utils-1.0.18.ori/alsactl/Makefile.in	Wed Oct 29 13:48:04 2008
+++ alsa-utils-1.0.18/alsactl/Makefile.in	Wed Dec 10 22:47:53 2008
@@ -204,7 +204,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = init
-man_MANS = alsactl.1 alsactl_init.7
+#man_MANS = alsactl.1 alsactl_init.7
 EXTRA_DIST = alsactl.1 alsactl_init.xml
 alsactl_SOURCES = alsactl.c state.c utils.c init_parse.c
 noinst_HEADERS = alsactl.h list.h init_sysdeps.c init_utils_string.c init_utils_run.c init_sysfs.c
@@ -659,8 +659,8 @@
 	uninstall-man1 uninstall-man7 uninstall-sbinPROGRAMS
 
 
-%.7: %.xml
-	xmlto man $?
+#%.7: %.xml
+#	xmlto man $?
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
