--- plugin-sysstat/lxqtsysstat.cpp
+++ plugin-sysstat/lxqtsysstat.cpp
@@ -28,9 +28,9 @@
 #include "lxqtsysstat.h"
 #include "lxqtsysstatutils.h"
 
-#include <SysStat/CpuStat>
-#include <SysStat/MemStat>
-#include <SysStat/NetStat>
+#include <sysstat/cpustat.h>
+#include <sysstat/memstat.h>
+#include <sysstat/netstat.h>
 
 #include <QTimer>
 #include <qmath.h>

--- plugin-sysstat/lxqtsysstatconfiguration.cpp
+++ plugin-sysstat/lxqtsysstatconfiguration.cpp
@@ -31,9 +31,9 @@
 #include "lxqtsysstatutils.h"
 #include "lxqtsysstatcolours.h"
 
-#include <SysStat/CpuStat>
-#include <SysStat/MemStat>
-#include <SysStat/NetStat>
+#include <sysstat/cpustat.h>
+#include <sysstat/memstat.h>
+#include <sysstat/netstat.h>
 
 
 LxQtSysStatConfiguration::LxQtSysStatConfiguration(QSettings *settings, QWidget *parent) :

