--- src/filename_list.cxx	Thu Sep 24 21:32:52 2009
+++ src/filename_list.cxx	Thu Sep 24 21:35:20 2009
@@ -67,7 +67,7 @@
   // The vast majority of UNIX systems want the sort function to have this
   // prototype, most likely so that it can be passed to qsort without any
   // changes:
-  int n = scandir(d, list, 0, (int(*)(const void*,const void*))sort);
+  int n = scandir(d, list, 0, (int(*)(const dirent **,const dirent **))sort);
 #else
   // This version is when we define our own scandir (WIN32 and perhaps
   // some Unix systems) and apparently on IRIX:
