--- src/fl_set_fonts_xft.cxx	2009-03-04 10:58:49.000000000 +0000
+++ src/fl_set_fonts_xft.cxx	2009-03-04 11:01:25.000000000 +0000
@@ -253,13 +253,13 @@
       // So the bit we want is up to the first comma - BUT some strings have
       // more than one name, separated by, guess what?, a comma...
       stop = start = first = 0;
-      stop = strchr((const char *)font, ',');
-      start = strchr((const char *)font, ':');
+      stop = strchr((char *)font, ',');
+      start = strchr((char *)font, ':');
       if ((stop) && (start) && (stop < start))
       {
         first = stop + 1; // discard first version of name
         // find first comma *after* the end of the name
-        stop = strchr((const char *)start, ',');
+        stop = strchr((char *)start, ',');
       }
       else
       {
