--- weechat-0.3.3/configure.orig
+++ weechat-0.3.3/configure
@@ -18828,7 +18828,7 @@
 weechat_libdir=${libdir}/weechat
 
 
-COMMON_CFLAGS="-Wall -W -Werror-implicit-function-declaration"
+COMMON_CFLAGS="-Wall -W"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether we have GNU assembler" >&5
 $as_echo_n "checking whether we have GNU assembler... " >&6; }
 GAS=`as --version < /dev/null 2>/dev/null | grep GNU`
