--- a/src/preferences.c
+++ b/src/preferences.c
@@ -74,10 +74,6 @@
 						   GTK_STOCK_CLOSE,
 						   GTK_RESPONSE_CLOSE);
 
-	gtk_dialog_add_button (GTK_DIALOG (pui->pbox),
-						   GTK_STOCK_HELP,
-						   GTK_RESPONSE_HELP);
-
 	g_signal_connect(G_OBJECT(pui->pbox), "response",
 					 G_CALLBACK(prefs_response_cb), pui);
 
--- a/src/ghex-ui.xml
+++ b/src/ghex-ui.new
@@ -215,7 +215,7 @@
 
 <submenu name="Help" _label="_Help">
 	
-	<menuitem name="Help" verb="" _label="_Contents"/>
+	
 
         <menuitem name="About" verb="" _label="_About"/>
 
