--- configure
+++ configure
@@ -808,7 +808,6 @@
 enable_maintainer_mode
 enable_dependency_tracking
 enable_nls
-enable_gtk3
 enable_shared
 enable_static
 with_pic
@@ -4798,20 +4797,7 @@
 fi
 
 
-# Check whether --enable-gtk3 was given.
-if test "${enable_gtk3+set}" = set; then :
-  enableval=$enable_gtk3; enable_gtk3="${enableval}"
-else
-  enable_gtk3="yes"
-
-fi
-
-
-if test x"$enable_gtk3" = x"yes"; then
-    pkg_modules="gtk+-3.0"
-else
     pkg_modules="gtk+-2.0 >= 2.16.0"
-fi
 
 
 
