--- etc/lxqt/lxqt.conf.orig
+++ etc/lxqt/lxqt.conf
@@ -3,2 +3,2 @@
-theme=light
-icon_theme=oxygen
+theme=flat
+icon_theme=SliTaz
--- etc/lxqt/session.conf.orig
+++ etc/lxqt/session.conf
@@ -8 +8 @@
-cursor_theme=whiteglass
+cursor_theme=SliTaz Polar
--- etc/pcmanfm-qt/lxqt/settings.conf.orig
+++ etc/pcmanfm-qt/lxqt/settings.conf
@@ -2,2 +2,2 @@
-IconThemeName=elementary
-SuCommand=gksu %s
+IconThemeName=SliTaz
+SuCommand=subox %s
@@ -18 +18 @@
-Wallpaper=/usr/share/lxqt/themes/light/simple_blue_widescreen.png
+Wallpaper=/usr/share/lxqt/themes/flat/flat.png
