--- src/spawn.c.orig
+++ src/spawn.c
@@ -411,8 +411,8 @@
 		{"View EXIF data (leafpad)", "exif %f | leafpad"},
 		{"View filesystem data (xterm)", "xterm -hold -e ls -l %f"},
 		{"Edit in Gimp", "gimp %f"},
-		{"View in GQview", "gqview %f"},
-		{"Print image", "kprinter %f"},
+		{"View in GPicView", "gpicview %f"},
+		{"Print image", "yad --print --type=IMAGE --print-add-preview --filename=%f"},
 		{"Email image", "seamonkey -compose attachment=file://%f"},
 		{"Send image to Firefox", "firefox %f"},
 		{"Send image to OpenOffice", "soffice %f"},
