diff -ur libofx-0.9.1-orig/lib/ofx_container_transaction.cpp libofx-0.9.1-patched/lib/ofx_container_transaction.cpp
--- libofx-0.9.1-orig/lib/ofx_container_transaction.cpp	2007-11-09 23:13:21.000000000 +0000
+++ libofx-0.9.1-patched/lib/ofx_container_transaction.cpp	2009-11-01 06:29:05.536051464 +0000
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_containers.hh"
diff -ur libofx-0.9.1-orig/lib/ofx_request.cpp libofx-0.9.1-patched/lib/ofx_request.cpp
--- libofx-0.9.1-orig/lib/ofx_request.cpp	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1-patched/lib/ofx_request.cpp	2009-11-01 06:29:22.034801091 +0000
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstring>
 #include "messages.hh"
 #include "libofx.h"
 #include "ofx_request.hh"
diff -ur libofx-0.9.1-orig/lib/ofx_request_accountinfo.cpp libofx-0.9.1-patched/lib/ofx_request_accountinfo.cpp
--- libofx-0.9.1-orig/lib/ofx_request_accountinfo.cpp	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1-patched/lib/ofx_request_accountinfo.cpp	2009-11-01 06:29:38.054801212 +0000
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
 
diff -ur libofx-0.9.1-orig/lib/ofx_request_statement.cpp libofx-0.9.1-patched/lib/ofx_request_statement.cpp
--- libofx-0.9.1-orig/lib/ofx_request_statement.cpp	2007-10-28 03:01:38.000000000 +0000
+++ libofx-0.9.1-patched/lib/ofx_request_statement.cpp	2009-11-01 06:29:54.464800978 +0000
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstdlib>
 #include "libofx.h"
 #include "ofx_utilities.hh"
 #include "ofx_request_statement.hh"
diff -ur libofx-0.9.1-orig/lib/ofx_utilities.cpp libofx-0.9.1-patched/lib/ofx_utilities.cpp
--- libofx-0.9.1-orig/lib/ofx_utilities.cpp	2007-10-27 13:15:58.000000000 +0100
+++ libofx-0.9.1-patched/lib/ofx_utilities.cpp	2009-11-01 06:28:38.574801207 +0000
@@ -18,10 +18,11 @@
 #include <config.h>
 #include <iostream>
 #include <assert.h>
+#include <cstdlib>
 
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
-#include <time.h>
+#include <ctime>
 #include <string>
 #include <locale.h>
 #include "messages.hh"
diff -ur libofx-0.9.1-orig/ofxdump/ofxdump.cpp libofx-0.9.1-patched/ofxdump/ofxdump.cpp
--- libofx-0.9.1-orig/ofxdump/ofxdump.cpp	2006-08-02 05:19:22.000000000 +0100
+++ libofx-0.9.1-patched/ofxdump/ofxdump.cpp	2009-11-01 06:31:23.564801519 +0000
@@ -27,12 +27,14 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
+#include <config.h>		/* Include config constants, e.g., VERSION TF */
 #include <iostream>
 #include <iomanip>
 #include <string>
 #include "libofx.h"
-#include <stdio.h>		/* for printf() */
-#include <config.h>		/* Include config constants, e.g., VERSION TF */
+#include <cstdlib>
+#include <cstdio>		/* for printf() */
+#include <cstring>
 #include <errno.h>
 
 #include "cmdline.h" /* Gengetopt generated parser */

 	  	 
