--- scim-pinyin-0.5.91.orig/src/scim_pinyin.cpp	Mon Aug  8 14:11:16 2005
+++ scim-pinyin-0.5.91/src/scim_pinyin.cpp	Mon Jun 22 12:24:33 2009
@@ -30,6 +30,7 @@
 
 #include <scim.h>
 #include "scim_pinyin.h"
+#include <cstring>
 
 // Internal functions
 static int
--- scim-pinyin-0.5.91.orig/src/scim_phrase.cpp	Thu Jan  6 21:30:58 2005
+++ scim-pinyin-0.5.91/src/scim_phrase.cpp	Mon Jun 22 12:25:33 2009
@@ -30,6 +30,7 @@
 #include <scim.h>
 #include "scim_pinyin_private.h"
 #include "scim_phrase.h"
+#include <cstring>
 
 static const char scim_phrase_lib_text_header [] = "SCIM_Phrase_Library_TEXT";
 static const char scim_phrase_lib_binary_header [] = "SCIM_Phrase_Library_BINARY";
--- scim-pinyin-0.5.91.orig/src/scim_pinyin_phrase.cpp	Sat Aug  6 20:06:49 2005
+++ scim-pinyin-0.5.91/src/scim_pinyin_phrase.cpp	Mon Jun 22 12:26:12 2009
@@ -31,6 +31,7 @@
 #include "scim_phrase.h"
 #include "scim_pinyin.h"
 #include "scim_pinyin_phrase.h"
+#include <cstring>
 
 static const char scim_pinyin_lib_text_header [] = "SCIM_Pinyin_Library_TEXT";
 static const char scim_pinyin_lib_binary_header [] = "SCIM_Pinyin_Library_BINARY";
--- scim-pinyin-0.5.91.orig/src/scim_special_table.cpp	Thu Jan  6 21:31:02 2005
+++ scim-pinyin-0.5.91/src/scim_special_table.cpp	Mon Jun 22 12:27:10 2009
@@ -24,6 +24,7 @@
 #include <scim.h>
 #include "scim_pinyin_private.h"
 #include "scim_special_table.h"
+#include <cstring>
 
 static inline String
 _trim_blank (const String &str)
