--- emesene-1.0.1/emesene.orig	Fri Jun 19 13:58:30 2009
+++ emesene-1.0.1/emesene	Fri Jun 19 13:59:15 2009
@@ -25,6 +25,22 @@
 except:
     pass
 
+# for slitaz openbox WM
+import os
+if not os.environ.has_key("KDE_FULL_SESSION") and \
+    not os.environ.has_key("DESKTOP_SESSION") and \
+    not os.environ.has_key("GNOME_DESKTOP_SESSION_ID"):
+    if not os.environ.has_key("DESKTOP_LAUNCH"):
+        if os.path.isfile('/usr/bin/firefox'):
+            browser = '/usr/bin/firefox '
+        elif os.path.isfile('/usr/bin/opera'):
+            browser = '/usr/bin/opera '
+        elif os.path.isfile('/usr/bin/midori'):
+            browser = '/usr/bin/midori '
+
+        if browser:
+            os.environ["DESKTOP_LAUNCH"] = browser
+
 try:
     from emesene import Controller
 except ImportError:
