--- gtk/Makefile.am
+++ gtk/Makefile.am
@@ -1357,10 +1357,6 @@
         fi &&								\
 	for i in 16 24; do						\
 	   (cd stock-icons/$$i						\
-	   && $(RM) document-save.png 					\
-	   && $(LN_S) media-floppy.png document-save.png 		\
-	   && $(RM) folder-remote.png 					\
-	   && $(LN_S) folder.png folder-remote.png 			\
 	   && $(RM) go-first-rtl.png 					\
 	   && $(LN_S) go-last-ltr.png go-first-rtl.png 			\
 	   && $(RM) go-last-rtl.png 					\
@@ -1377,10 +1373,6 @@
 	   && $(LN_S) media-skip-forward-ltr.png media-skip-backward-rtl.png \
 	   && $(RM) media-skip-forward-rtl.png 				\
 	   && $(LN_S) media-skip-backward-ltr.png media-skip-forward-rtl.png \
-	   && $(RM) user-desktop.png 					\
-	   && $(LN_S) folder.png user-desktop.png 			\
-	   && $(RM) user-home.png 					\
-	   && $(LN_S) folder.png user-home.png 				\
 	) done 								\
 	&& touch stamp-icons
 
