--- Makefile	Thu Sep 18 04:26:56 2008
+++ Makefile	Thu Feb 26 01:16:24 2009
@@ -136,16 +136,7 @@
 endif	
 else
 	cp -f os/linux/Makefile.6 $(RT28xx_DIR)/os/linux/Makefile
-	make  -C  $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules
-ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.ko /tftpboot
-else	
-ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.ko /tftpboot
-else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.ko /tftpboot
-endif	
-endif
+	make  -C  $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules	
 endif
 
 clean:
