--- os/linux/config.mk	Thu Sep 18 05:27:46 2008
+++ os/linux/config.mk	Sat Mar  7 19:39:13 2009
@@ -5,10 +5,10 @@
 HAS_28xx_QA=n
 
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
@@ -39,6 +39,8 @@
 #Support features of 802.11n
 HAS_DOT11_N_SUPPORT=y
 
+#Support EeePC Special setting
+HAS_EEEPC_SPECIAL_SETTING=y
 
 #################################################
 
@@ -49,6 +51,10 @@
 
 
 #################################################
+
+ifeq ($(HAS_EEEPC_SPECIAL_SETTING),y)
+WFLAGS += -DEEEPC_SPECIAL_SETTING
+endif
 
 #ifdef CONFIG_STA_SUPPORT
 # config for STA mode
