--- configure.in
+++ configure.in
@@ -94,21 +94,15 @@
 
 dnl Imlib2 detection
 
-AC_PATH_GENERIC(imlib2, , [
-    AC_SUBST(IMLIB_LIBS)
-   AC_SUBST(IMLIB_CXXFLAGS) ],
-  AC_MSG_ERROR(Cannot find imlib2: Is imlib2-config in the path?
-               You need Imlib2 to build Idesk.  Verify that you have Imlib2-dev))
-
-dnl the above doesn't work for some reason :/
-IMLIB_LIBS=`imlib2-config --libs`
-IMLIB_CFLAGS=`imlib2-config --cflags`
-AC_SUBST(IMLIB_LIBS)
-AC_SUBST(IMLIB_CXXFLAGS)
-
-CXXFLAGS="$CXXFLAGS $IMLIB_CXXFLAGS"
-LIBS="$LIBS $IMLIB_LIBS"
+PKG_CHECK_MODULES(IMLIB2, [imlib2], [imlib2_pkgconfig=yes], [imlib2_pkgconfig=no])
+if test "x$imlib2_pkgconfig" = "xyes"; then
+   AC_DEFINE(HAVE_IMLIB2, 1, [Define to 1 if you have imlib2])
+fi
+AC_SUBST(IMLIB2_LIBS)
+AC_SUBST(IMLIB2_CXXFLAGS)
 
+CXXFLAGS="$CXXFLAGS $IMLIB2_CXXFLAGS"
+LIBS="$LIBS $IMLIB2_LIBS"
 
 dnl Check for libsn Startup Notificacion options
 STARTUP=""
