gawk raise a warning with some scripts.

--- a/modules/compressor
+++ b/modules/compressor
@@ -376,7 +376,7 @@
 		echo >> "$tmpfile"
 
 		# get Plural-Forms
-		awk '
+		busybox awk '
 		BEGIN { skip = ""; }
 		{
 			if (! skip) {
@@ -394,7 +394,7 @@
 		fi
 
 		# main
-		awk -v pf="$(cat "$tmpfile.pf")" '
+		busybox awk -v pf="$(cat "$tmpfile.pf")" '
 		function clean() {
 			mode = msgctxt = msgid = msgid_plural = msgstr = msgstr0 = msgstr1 = msgstr2 = msgstr3 = msgstr4 = msgstr5 = "";
 		}
