The wanted package *must* have the same version
--- a/cook
+++ b/cook
@@ -167,6 +167,9 @@
 		src="$basesrc/$WANTED-$VERSION"
 		install="$WOK/$WANTED/install"
 		wanted_stuff="$WOK/$WANTED/stuff"
+		wanted_version="$(. $WOK/$WANTED/receipt; echo $VERSION)"
+		[ $wanted_version = $VERSION ] ||
+			(_ 'ERROR: %s' "wanted package version $wanted_version" && exit 1)
 	fi
 
 	[ -n "$SOURCE" ] && source_stuff="$WOK/$SOURCE/stuff"
