--- a/gksu-migrate-conf.sh
+++ b/gksu-migrate-conf.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 set_default ()
 {

# FIXME: gksu is incompatible with /usr/bin/terminal!
# Using sakura here instead.

--- a/gksu.desktop
+++ b/gksu.desktop
@@ -41,7 +41,7 @@
 Comment[th]=เปิดเทอร์มินัลในนามผู้ใช้ root โดยใช้ gksu ถามรหัสผ่าน
 Comment[zh_CN]=以 root 用户身份打开一个终端，使用 gksu 询问密码
 Encoding=UTF-8
-Exec=gksu /usr/bin/x-terminal-emulator
+Exec=gksu sakura
 Icon=gksu-root-terminal
 StartupNotify=true
 Terminal=false
