--- deps/v8/SConstruct
+++ deps/v8/SConstruct
@@ -80,8 +80,8 @@
   },
   'gcc': {
     'all': {
-      'CCFLAGS':      ['$DIALECTFLAGS', '$WARNINGFLAGS'],
-      'CXXFLAGS':     ['-fno-rtti', '-fno-exceptions'],
+      'CCFLAGS':      ['$DIALECTFLAGS', '$WARNINGFLAGS', '-march=armv6'],
+      'CXXFLAGS':     ['-fno-rtti', '-fno-exceptions', '-march=armv6'],
     },
     'visibility:hidden': {
       # Use visibility=default to disable this.
