Index: trunk/Source/WebCore/xml/XPathParser.cpp
===================================================================
--- trunk/Source/WebCore/xml/XPathParser.cpp	(revision 97771)
+++ trunk/Source/WebCore/xml/XPathParser.cpp	(revision 124099)
@@ -33,22 +33,19 @@
 #include "XPathException.h"
 #include "XPathNSResolver.h"
+#include "XPathPath.h"
 #include "XPathStep.h"
 #include <wtf/StdLibExtras.h>
 #include <wtf/text/StringHash.h>
 
-int xpathyyparse(void*);
-
+using namespace WebCore;
 using namespace WTF;
 using namespace Unicode;
-
-namespace WebCore {
-namespace XPath {
-
-class LocationPath;
-
-#include "XPathGrammar.h"    
+using namespace XPath;
+
+extern int xpathyyparse(WebCore::XPath::Parser*);
+#include "XPathGrammar.h"
 
 Parser* Parser::currentParser = 0;
-    
+
 enum XMLCat { NameStart, NameCont, NotPartOfName };
 
@@ -631,4 +628,2 @@
 }
 
-}
-}
