--- src/pnm.c.orig	2018-10-15 18:47:50.000000000 +0100
+++ src/pnm.c	2019-03-07 14:00:14.742892543 +0100
@@ -467,7 +467,7 @@ void addrgb(unsigned char rgb[3], int sr
    else          rgb[i]=((rgb[i]+add[i]>255)?255:rgb[i]+add[i]);
 }
 /*
- * pgmtoppm or pnmtopng, use last 3 bits for farbcoding 
+ * pgmtoppm or pnm2png, use last 3 bits for farbcoding 
  * replaces old writebmp variant
  *  2018-10 add $opt to control coloring
  */
@@ -479,11 +479,11 @@ int writeppm(char *nam, pix *p, int opt)
 #ifdef HAVE_POPEN
   /* be sure that nam contains hacker code like "dummy | rm -rf *" */
   if (!f1) {
-    strncpy(buf,"pnmtopng > ",12);  /* no spaces within filenames allowed! */
+    strncpy(buf,"pnm2png > ",12);  /* no spaces within filenames allowed! */
     strncpy(buf+11,nam,111); buf[123]=0;
     strncpy(buf+strlen(buf),".png",5);
     /* we dont care about win "wb" here, never debug on win systems */
-    f1 = popen(buf, "w"); if(f1) f1t=1; else E0("popen pnmtopng");
+    f1 = popen(buf, "w"); if(f1) f1t=1; else E0("popen pnm2png");
   }
   if (!f1) {
     strncpy(buf,"gzip -c > ",11);
