--- src/ptk/ptk-file-archiver.c	Sun May 25 14:59:12 2008
+++ src/ptk/ptk-file-archiver.c	Thu Jun 25 23:28:10 2009
@@ -10,6 +10,8 @@
 *
 */

+/* 06-25-2009 Erjo: Add support for tazpkg */
+
 #include <glib/gi18n.h>
 #include <string.h>

@@ -61,6 +63,12 @@
             "tar -xvf",
             ".tar", TRUE
         },
+		{
+            "application/x-tazpkg",
+            "tazpkg pack",
+            "tazpkg extract",
+            ".tazpkg", FALSE
+        }
         /*
         {
             "application/x-rar",
