--- a/src/sakura.c
+++ b/src/sakura.c
@@ -196,7 +196,7 @@
 	gint scrollbar_key;
 	gint fullscreen_key;
 	GRegex *http_regexp;
-	char *argv[2];
+	char *argv[3];
 } sakura;
 
 struct terminal {
@@ -1825,12 +1825,13 @@
 	}
 
 	/* Set argv for forked childs */
+	sakura.argv[0]=g_strdup(g_getenv("SHELL"));
 	if (option_login) {
-		sakura.argv[0]=g_strdup_printf("-%s", g_getenv("SHELL"));
+		sakura.argv[1]=g_strdup("-l");
 	} else {
-		sakura.argv[0]=g_strdup(g_getenv("SHELL"));
+		sakura.argv[1]=NULL;
 	}
-	sakura.argv[1]=NULL;
+	sakura.argv[2]=NULL;
 
 	if (option_title) {
 		gtk_window_set_title(GTK_WINDOW(sakura.main_window), option_title);
