Set Freedesktop standard icon

--- a/src/sakura.c
+++ b/src/sakura.c
@@ -210,7 +210,7 @@
 };
 
 
-#define ICON_FILE "terminal-tango.svg"
+#define ICON_NAME "utilities-terminal"
 #define SCROLL_LINES 4096
 #define HTTP_REGEXP "(ftp|http)s?://[-a-zA-Z0-9.?$%&/=_~#.,:;+]*"
 #define CONFIGFILE "sakura.conf"
@@ -1803,7 +1803,7 @@
 
 	sakura.main_window=gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(sakura.main_window), "sakura");
-	gtk_window_set_icon_from_file(GTK_WINDOW(sakura.main_window), DATADIR "/pixmaps/" ICON_FILE, &gerror);
+	gtk_window_set_icon_name(GTK_WINDOW(sakura.main_window), ICON_NAME);
 	/* Default terminal size*/
 	sakura.columns = DEFAULT_COLUMNS;
 	sakura.rows = DEFAULT_ROWS;
